pkg/manager/appliance.go

```diff
@@ -148,6 +148,22 @@ func (a *Appliance) AddBlade(ctx context.Context, c *openapi.Credentials) (*Blad
 
 		newErr := fmt.Errorf("appliance [%s] new blade object creation failure: %w", a.Id, err)
 		logger.Error(newErr, "failure: add blade")
+
+		// Continue adding the failed blade to the datastore, but update the connection status to unavailable
+		applianceDatum, _ := datastore.DStore().GetDataStore().GetApplianceDatumById(a.Id)
+		applianceDatum.AddBladeDatum(c)
+		OfflineBlade, _ := applianceDatum.GetBladeDatumById(ctx, c.CustomId)
+		OfflineBlade.ConnectionStatus = common.UNAVAILABLE
+		datastore.DStore().Store()
+		
+		var newBlade = &Blade{
+			Id:          OfflineBlade.Credentials.CustomId,
+			Uri:         GetCfmUriBladeId(a.Id, OfflineBlade.Credentials.CustomId),
+			Status:      OfflineBlade.ConnectionStatus,
+			ApplianceId: a.Id,
+		}
+		a.Blades[newBlade.Id] = newBlade
+
```

Comment:
Hmmm, I see what you need here, but this won't quite work.
There's a couple issues here. We should talk offline about this.

The UNAVAILABLE state happens when you discover a blade with `cma=true` but can NOT talk to it (i.e." Something is wrong with the BMC or the blade firmware).  
This means that:
1.) the `ops.CreateSession() above will fail and never get here. and
2.) we don't want this to run this code here every time someone fails to add a new blade (via the webui \ cli)
We only want to run this code during discover, when the datastore is attempting a reload.